ipi/engine/motion/driven_dynamics.py

```diff
@@ -0,0 +1,177 @@
+
+import numpy as np
+
+from ipi.utils.depend import *
+from ipi.engine.motion.eda import EDA
+from ipi.utils.units import Constants
+from ipi.engine.motion.dynamics import NVEIntegrator, DummyIntegrator, Dynamics
+
+
+class DrivenDynamics(Dynamics):
+    """self (path integral) molecular dynamics class.
+
+    Gives the standard methods and attributes needed in all the
+    dynamics classes.
+
+    Attributes:
+        beads: A beads object giving the atoms positions.
+        cell: A cell object giving the system box.
+        forces: A forces object giving the virial and the forces acting on
+            each bead.
+        prng: A random number generator object.
+        nm: An object which does the normal modes transformation.
+
+    Depend objects:
+        econs: The conserved energy quantity appropriate to the given
+            ensemble. Depends on the various energy terms which make it up,
+            which are different depending on the ensemble.he
+        temp: The system temperature.
+        dt: The timestep for the algorithms.
+        ntemp: The simulation temperature. Will be nbeads times higher than
+            the system temperature as PIMD calculations are done at this
+            effective classical temperature.
+    """
+
+    def __init__(
+        self,
+        efield=None,
+        bec=None,
+        *argc,
+        **argv,
+    ):
+        """Initialises a "dynamics" motion object.
+
+        Args:
+            dt: The timestep of the simulation algorithms.
+            fixcom: An optional boolean which decides whether the centre of mass
+                motion will be constrained or not. Defaults to False.
+        """
+
+        super().__init__(*argc, **argv)
+
+        if self.enstype == "eda-nve":
+            # NVE integrator with an external time-dependent driving (electric field)
+            self.integrator = EDANVEIntegrator()
+        else:
+            self.integrator = DummyIntegrator()
+
+        # if the dynamics is driven, allocate necessary objects
+        self.efield = efield
+        self.bec = bec
+        self.eda = EDA(self.efield, self.bec)
+
+    def bind(self, ens, beads, nm, cell, bforce, prng, omaker):
+
+        super().bind(ens, beads, nm, cell, bforce, prng, omaker)
+
+        self.eda.bind(self.ensemble, self.enstype)
+
+        # now that the timesteps are decided, we proceed to bind the integrator.
+        self.integrator.bind(self)
+
+        # applies constraints immediately after initialization.
+        self.integrator.pconstraints()
+
+    def step(self, step=None):
+        """Advances the dynamics by one time step"""
+
+        super().step(step)
+        # self.integrator.step(step)
+        # self.ensemble.time += self.dt  # increments internal time
+
+        # Check that these variable are the same.
+        # If they are not the same, then there is a bug in the code
+        dt = abs(self.ensemble.time - self.integrator.mts_time)
+        if dt > 1e-12:
```

Comment:
They should be always the same at that point of the code.
In case of further changes to the code, I think this a simple way to check if the time-dependent integrators are working properly. 
I can remove or comment these lines, but I think it's a simple check, usefull to future developers.